pkg/controllers/user/rbac/roletemplate_handler.go

```diff
@@ -92,114 +88,51 @@ func (c *rtLifecycle) syncRT(template *v3.RoleTemplate, usedInProjects bool, prt
 		}
 	}
 
+	for _, rtName := range template.RoleTemplateNames {
+		rt, err := c.m.rtLister.Get("", rtName)
+		if err != nil {
+			return err
+		}
+		roles[rtName] = rt
```

Comment:
what about the parents of parents and so on? in prtb handler we are gathering them recursively, should we be doing the same here?

Reply:
syncRT also calles gatherRoles which gathers them recursively https://github.com/rancher/rancher/pull/24760/files#diff-e5bf9d33608d38443a803c450bc5a590L68

So this for loop isn't needed, removing it

Reply:
perfect